packages/playwright-crawler/src/internals/adaptive-playwright-crawler/adaptive-playwright-crawler.ts

```diff
@@ -0,0 +1,229 @@
+import { addTimeoutToPromise } from '@apify/timeout';
+import { MemoryStorage } from '@crawlee/memory-storage';
+import type { BatchAddRequestsResult } from '@crawlee/types';
+import { extractUrlsFromCheerio } from '@crawlee/utils';
+import { load, type CheerioAPI } from 'cheerio';
+
+import { calculateChangeRatio, type RenderingType } from './rendering-type-detection';
+import { RenderingTypePredictor } from './rendering-type-prediction';
+import type { Configuration, EnqueueLinksOptions, PlaywrightCrawlingContext, PlaywrightCrawlerOptions, BasicCrawlerOptions, Dictionary } from '../..';
+import { PlaywrightCrawler } from '../..';
```

Comment:
Just one note, this is generally a bad practice that can introduce import cycles, as you import from the index file which will later on re-export this file too. Things like this often work during development and break later on when a package is published or when downstream users try to bundle it.